### ISSUE TITLE
Target-Sensitive Memory Networks for Aspect Sentiment Classification

### ISSUE THREAD

**Issue by wwacky** [wwacky/survey#1]
## 一言でいうと
レビューに含まれるaspectの極性を、前後のcontextから判別するMemory Networkを構築。contextによってはaspectによって極性が変わる("高い"など)ので、aspectとcontextの交互作用を組み込んだNetworkを提案し、SemEval Task 2014のデータで有効性を検証した。

### 論文リンク
http://www.aclweb.org/anthology/P18-1088

### 著者/所属機関
- Shuai Wang (Department of Computer Science, University of Illinois at Chicago, USA)
- Sahisnu Mazumder (Department of Computer Science, University of Illinois at Chicago, USA)
- Bing Liu (Department of Computer Science, University of Illinois at Chicago, USA)
- Mianwei Zhou (Plus.AI, USA https://plus.ai/en/about/)
- Yi Chang (Artificial Intelligence School, Jilin University, China)

### 投稿日付
2018-07-12 ACL2018

## 概要
- aspectとcontext (aspect前後の単語列)から、aspectに対する極性を分類する
- aspectは事前に特定されているものとする
- contextに対してAttentionを適用する
- SemEval Task2014のデータセットで検証し、F1が比較対象より高いことを確認
- "高い"というcontextが"価格"というaspectと"解像度"というaspectで極性が変わるように学習していることを確認

## 新規性・差分
DeepNetworkで、aspectとcontextの交互作用を組み込んだモデルを提案。

## 手法

- ベースとするモデル
  - Input Representation
  - Attention
    - word xに対するattention weight αを設定する
  - Output Representation
  - Sentiment Score(or Logit)
    - positive, neutral, negativeの3クラスに対してスコアが計算される
    - softmaxで3クラスそれぞれの確率値を計算する

このままだと線形モデルのため、Aspectとcontextの交互作用が考慮されていない。
aspectが"price"と"resolution"ではhighによって極性がPositiveかNegativeか変わる

- Proposed Approches
  - (NP) Non-linear Projection
    - attention weighted context(αc) とaspect(v)を足した後にtanhで射影する
  - (CNP) Contextual Non-linear Projection
    - context(c)とaspect(v)の合計をtanhで射影した後にattention weight(α)を積算する
  - (IT) Interaction Term
    - contextとaspectのembedding vectorの積からなる行列Dを新たに生成し、continuous vector(c)と合算する
    - 行列Dをtarget-context sentiment (TCS) interactionと呼んでいる 
  - (CI) Coupled Interaction
    - ITで作成した行列Dに新たなembedding layer(e)を積算する
  - (JCI) Joint Coupled Interaction
    - CIのembedding layer(e)をcontinuous vector(c)に変更し、パラメータの数を減らす
  - (JPI) Joint Projected Interaction
    - contextとTCSの行列Dを別空間で計算させるために2つの項に分ける。理由は良くわからない…
  - 提案手法の損失関数はすべてcross entropy loss

## 結果
SemEval Task 2014(LaptopとRestaurantのレビュー)で検証。
![image](https://user-images.githubusercontent.com/4422849/51440619-453dc580-1d0c-11e9-9634-45daabe44381.png)
※Target-Sensitive Memory Networks for Aspect Sentiment Classificationより引用

![image](https://user-images.githubusercontent.com/4422849/51440662-db71eb80-1d0c-11e9-8c2c-899275b094fc.png)
※Target-Sensitive Memory Networks for Aspect Sentiment Classificationより引用

MN, AMN, BL-MN,AE-LSTM, ATAE-LSTMは比較対象の既存手法
MacroはNegative, Neutral, Positiveの全クラスをまとめてF1を算出したもの。
MicroはAccuracyのことらしい。

提案手法のJCIが一番バランス良さそうに見える。

![image](https://user-images.githubusercontent.com/4422849/51440895-1bd26900-1d0f-11e9-851d-5b3cf8a5c1f7.png)
※Target-Sensitive Memory Networks for Aspect Sentiment Classificationより引用

MNの中身はBL-MN、TMNの中身はJCI。他の手法でも大体傾向は同じ。
**提案手法(TMN)だと、higherが価格の時はNegativeで、解像度の時はPositiveになっている。
higher以外にoverでも確認したが、priceの時はNegativeでdinnerの時はNeutralになっていた。**

## コメント
Table5から'higher'の極性への影響を正しく学習しているので、正しく学習できているケースもありDeepNetworkでモデリングする場合は参考にできそう。
ただし、Table3を見ると交互作用が無い比較手法とF値が最大2ポイント程度しか変わらないようなので、大きな精度向上にはつながらないのかもしれない（交互作用以外にもっと大きな問題がある？）

Related Worksで参考になりそうな論文。
sentiment analysisのsurvey : [Deep Learning for Sentiment Analysis : A Survey](https://arxiv.org/pdf/1801.07883.pdf)

sentiment analysisでattentionを用いているもの(aspect detectionに有用)
- [Recurrent Attention Network on Memory for Aspect Sentiment Analysis](https://www.cs.cmu.edu/~lbing/pub/emnlp17_aspect_sentiment.pdf)
- [Attention Modeling for Targeted Sentiment](http://leoncrashcode.github.io/Documents/EACL2017.pdf)
- [Interactive Attention Networks for Aspect-Level Sentiment Classification](https://www.ijcai.org/proceedings/2017/0568.pdf)

## 略語
頭字語 | 意味
--- | ---
TMNs | Target-sensitive Memory Networks
ASC | Aspect sentiment classification
TCS | Target Context Sentiment

**Comment by wwacky** [comment]
発表の動画
https://vimeo.com/285801326
